### PR TITLE
feat: azureaisearch support collection string

### PR DESCRIPTION
**Pull request by EmanuelCampos**

### CODE REVIEW

File: llama-index-integrations/vector_stores/llama-index-vector-stores-azureaisearch/llama_index/vector_stores/azureaisearch/base.py [hunk 1]
```diff
@@ -813,7 +819,7 @@ def _create_index_document(self, node: BaseNode) -> Dict[str, Any]:
         node_metadata = node_to_metadata_dict(
             node,
             remove_text=True,
-            flat_metadata=self.flat_metadata,
+            flat_metadata=False,
```
Comment (logan-markewich) [inline llama-index-integrations/vector_stores/llama-index-vector-stores-azureaisearch/llama_index/vector_stores/azureaisearch/base.py:822]: We should still use self.flat_metadata, but just change it to false on the class attribute?

File: llama-index-integrations/vector_stores/llama-index-vector-stores-azureaisearch/llama_index/vector_stores/azureaisearch/base.py [hunk 2]
```diff
@@ -869,6 +875,18 @@ async def adelete(self, ref_doc_id: str, **delete_kwargs: Any) -> None:
...
+                    f"t eq '{value}'" if isinstance(value, str) else f"t eq {value}"
+                    for value in f.value
+                ]
+            )
+            # Construct the filter query using the any function
+            return f"allowed_users_ids/any(t: {value_str})"
+
         for f in metadata_filters.legacy_filters():
             if not isinstance(f, ExactMatchFilter):
                 raise NotImplementedError(
```
Comment (logan-markewich) [inline llama-index-integrations/vector_stores/llama-index-vector-stores-azureaisearch/llama_index/vector_stores/azureaisearch/base.py:892]: Should we remove this block? We either handle only ExactMatchFilters or we handle all filters

Above, it looks like you are still only handling exact match filters?

**Review by sourabhdesai — APPROVED** [review]

File: llama-index-integrations/vector_stores/llama-index-vector-stores-azureaisearch/llama_index/vector_stores/azureaisearch/base.py (hunk 1 above)
Comment (nerdai) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-azureaisearch/llama_index/vector_stores/azureaisearch/base.py]: @EmanuelCampos was this resolved?

Comment (EmanuelCampos) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-azureaisearch/llama_index/vector_stores/azureaisearch/base.py]: @nerdai yes!

Comment (nerdai) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-azureaisearch/llama_index/vector_stores/azureaisearch/base.py]: thanks! :)

**Review by nerdai — APPROVED** [review]
lgtm!

**Pull request merged**